recipe/build-arrow.sh

```diff
@@ -30,43 +30,71 @@ if [ "$(uname)" == "Linux" ]; then
   sed -ie 's;"--with-jemalloc-prefix\=je_arrow_";"--with-jemalloc-prefix\=je_arrow_" "--with-lg-page\=16";g' ../cmake_modules/ThirdpartyToolchain.cmake
 fi
 
-# we don't provide right now CUDA variant
-EXTRA_CMAKE_ARGS=" ${EXTRA_CMAKE_ARGS} -DARROW_CUDA=OFF"
+# Enable CUDA support
+if [ "${build_type}" = "cuda" ]; then
+  if [[ -z "${CUDA_HOME+x}" ]]
+    then
+        echo "cuda version=${cudatoolkit} CUDA_HOME=$CUDA_HOME"
+        CUDA_GDB_EXECUTABLE=$(which cuda-gdb || exit 0)
+        if [[ -n "$CUDA_GDB_EXECUTABLE" ]]
+        then
+            CUDA_HOME=$(dirname $(dirname $CUDA_GDB_EXECUTABLE))
+        else
+            echo "Cannot determine CUDA_HOME: cuda-gdb not in PATH"
+            return 1
+        fi
+    fi
+  EXTRA_CMAKE_ARGS=" ${EXTRA_CMAKE_ARGS} -DARROW_CUDA=ON -DCUDA_TOOLKIT_ROOT_DIR=${CUDA_HOME} -DCMAKE_LIBRARY_PATH=${CUDA_HOME}/lib64/stubs"
+else
+  EXTRA_CMAKE_ARGS=" ${EXTRA_CMAKE_ARGS} -DARROW_CUDA=OFF"
+fi
+# Disable Gandiva
 EXTRA_CMAKE_ARGS=" ${EXTRA_CMAKE_ARGS} -DARROW_GANDIVA=OFF"
 export BOOST_ROOT="${LIBRARY_PREFIX}"
 export Boost_ROOT="${LIBRARY_PREFIX}"
 
+#SIMD Settings
+if [[ "${target_platform}" == "linux-64" ]]; then
+  EXTRA_CMAKE_ARGS=" ${EXTRA_CMAKE_ARGS} -DARROW_SIMD_LEVEL=SSE4_2"
+fi
+if [[ "${target_platform}" == "linux-ppc64le" ]]; then
+  EXTRA_CMAKE_ARGS=" ${EXTRA_CMAKE_ARGS} -DARROW_ALTIVEC=ON"
+fi
+
 cmake \
     -DARROW_BOOST_USE_SHARED=ON \
     -DARROW_BUILD_BENCHMARKS=OFF \
     -DARROW_BUILD_STATIC=OFF \
     -DARROW_BUILD_TESTS=OFF \
     -DARROW_BUILD_UTILITIES=OFF \
     -DBUILD_SHARED_LIBS=ON \
-    -DARROW_SSE42=OFF \
-    -DCMAKE_BUILD_TYPE=release \
-    -DCMAKE_INSTALL_PREFIX=$PREFIX \
-    -DCMAKE_INSTALL_LIBDIR=$PREFIX/lib \
+    -DARROW_DATASET=ON \
```

Comment:
Previously, this was `-DARROW_DATASETS=ON` seems to be missing an S now.

Reply:
yeah - its supposed to not have the `S`:

https://github.com/apache/arrow/blob/apache-arrow-3.0.0/cpp/cmake_modules/FindArrowDataset.cmake#L50